File: 20240814/Q2.cpp
#include <iostream>
#include <string>

using namespace std;

string roman(int num) {
    int val[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    string syb[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
    
    string roman_num = "";
    
    for (int i = 0; i < 13; i++) {
        while (num >= val[i]) {
            roman_num += syb[i];
            num -= val[i];
        }
    }
    
    return roman_num;
}

int main() {
    int arabic_num;
    
    cout << "請輸入一個阿拉伯數字（1-3999）：";
    cin >> arabic_num;

    cout << "羅馬數字表示法: " << roman(arabic_num) << endl;
    return 0;
}

File: 20240814/Q2.py
def roman(num):
    dic = {
        1000: "M",
        900: "CM",
        500: "D",
        400: "CD",
        100: "C",
        90: "XC",
        50: "L",
        40: "XL",
        10: "X",
        9: "IX",
        5: "V",
        4: "IV",
        1: "I"
    }
    
    roman_num = ''

    for value in dic:
        while num >= value:
            roman_num += dic[value]
            num -= value

    return roman_num

arabic_num = int(input("請輸入一個阿拉伯數字（1-3999）："))
print(f"羅馬數字表示法: {roman(arabic_num)}")
